labelbox/data/annotation_types/metrics/base.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC
+from pydantic import ValidationError, confloat, BaseModel, validator
+from pydantic.error_wrappers import ErrorWrapper
+from typing import Dict, Optional, Any, Union
+
+ConfidenceValue = confloat(ge=0, le=1)
+
+MIN_CONFIDENCE_SCORES = 2
+MAX_CONFIDENCE_SCORES = 15
+
+
+class BaseMetric(BaseModel, ABC):
+    value: Union[Any, Dict[float, Any]]
+    feature_name: Optional[str] = None
+    subclass_name: Optional[str] = None
+    extra: Dict[str, Any] = {}
+
+    def dict(self, *args, **kwargs):
+        res = super().dict(*args, **kwargs)
+        return {k: v for k, v in res.items() if v is not None}
+
+    @validator('value')
+    def validate_value(cls, value):
+        if isinstance(value, Dict):
```

Comment:
should it throw an error if it isnt an instance of a `Dict`?

Reply:
no because it is valid to not be a dict. Just if it is, then we want to validate the confidence ranges.